jwst/pipeline/calwebb_coron3.py

```diff
@@ -142,18 +142,22 @@ def process(self, input):
 
         # Call the resample step to combine all the psf-subtracted target images
         result = self.resample(resample_input)
+        output_file = mk_prodname(self.output_dir, prod['name'], 'i2d')
 
-        # TEMPORARY HACK UNTIL RESAMPLE IS VIABLE
-        log.warning('Creating fake resample results until step is available')
-        result = datamodels.DrizProductModel(data=resample_input[0].data,
+        if result == resample_input:
+        # Resampling was skipped, 
+        #  yet we need to return a DrizProductModel, so...
+            log.warning('Creating fake resample results until step is available')
+            result = datamodels.DrizProductModel(data=resample_input[0].data,
                                              con=resample_input[0].dq,
                                              wht=resample_input[0].err)
-        result.update(resample_input[0])
-        output_file = mk_prodname(self.output_dir, prod['name'], 'i2d')
-        log.debug('Blending metadata for {}'.format(output_file))
-        blend.blendfitsdata(targ_files, result)
-        result.meta.asn.pool_name = asn['asn_pool']
-        result.meta.asn.table_name = input
+            result.update(resample_input[0])
+            # The resample step blends headers already...
+            log.debug('Blending metadata for {}'.format(output_file))
+            blend.blendfitsdata(targ_files, result)
+            result.meta.asn.pool_name = asn['asn_pool']
+            result.meta.asn.table_name = input
+
         result.meta.cal_step.outlier_detection = 'COMPLETE'
```

Comment:
Was this hardwired here because it was assumed that outlier_detection was being skipped? If so, now that outlier_detection has been enabled, the status should be set within outlier_detection and hence shouldn't be needed here anymore.

Reply:
It turns out `outlier_detection_step` does not update to "COMPLETE", so this is a change I've put into #1173 with an upcoming commit and merge.  We need `outlier_detection_step` to do this in order for the pipeline to know if it should write out the Level 2c product.  That logic is also coming in #1173.